### PR TITLE
Add DockerFile for ChromeDevTools

### PR DESCRIPTION
**Pull request by TimvdLippe**
Note that I was not able to verify it fully works, as it throws an
authentication error on typescript/typescript on the Docker Hub.

This is part of #39568

CC @weswigham

### CODE REVIEW

File: tests/cases/docker/chrome-devtools-frontend-next/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+RUN mkdir devtools
+WORKDIR devtools
+RUN fetch devtools-frontend
+WORKDIR devtools-frontend
+RUN gn gen out/Default
+COPY --from=typescript/typescript /typescript/typescript-*.tgz typescript.tgz
+RUN mkdir /typescript
+RUN tar -xzvf /typescript.tgz -C /typescript
+RUN ln -s /typescript/package ./node_modules/typescript
+RUN autoninja -C out/Default front_end
```
Comment (TimvdLippe) [inline tests/cases/docker/chrome-devtools-frontend-next/Dockerfile:14]: For some reason if this last step is a `CMD` then it doesn't do anything at all. I am not sure why, but the `RUN` works as expected.

File: tests/cases/docker/chrome-devtools-frontend-next/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FROM node:current
+RUN git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git /depot_tools
+ENV PATH=/depot_tools:$PATH
+WORKDIR /
+RUN mkdir devtools
+WORKDIR devtools
+RUN fetch devtools-frontend
```
Comment (TimvdLippe) [inline tests/cases/docker/chrome-devtools-frontend-next/Dockerfile:7]: The instructions here and below are taken from https://github.com/ChromeDevTools/devtools-frontend/blob/master/docs/workflows.md#standalone-checkout

**Comment by weswigham** [comment]
`typescript/typescript` is built locally from the `Dockerfile` in the repo root before running the docker tests. `gulp run tests --ru=docker --t=devtools` should work.

File: tests/cases/docker/chrome-devtools-frontend-next/Dockerfile (hunk 1 above)
Comment (weswigham) [reply tests/cases/docker/chrome-devtools-frontend-next/Dockerfile:14]: A docker file is split into two phases, build and run. All the `RUN`s build the FS in the `Dockerfile`. The `CMD` is what is executed (by default) when you actually _execute_ the built docker image. We use that here - only the `CMD` output is baseline'd.

Comment (TimvdLippe) [reply tests/cases/docker/chrome-devtools-frontend-next/Dockerfile:14]: Ah I see, thanks for the explanation.

**Comment by TimvdLippe** [comment]
I was able to make it run with:

```
npm run gulp -- runtests --ru=docker --t=chrome-devtools-frontend-next
```

I am not sure how best to make the error log as useful for you. The CMD now makes sure that into only shows any failures, if there are any. This should remove all noise from Ninja and produce a stable output.

**Comment by weswigham** [comment]
@typescript-bot user test this

**Comment by TimvdLippe** [comment]
@weswigham It's been a while since we worked on this PR. Could you let me know what the latest status update for it is?

**Pull request merged**